Server crash when opening timeline tab for big workflows

When opening the timeline tab (URL /workflows/argo/<workflow>?tab=timeline) for big workflows (~3k nodes),
Argo server crashes:


```
time="2021-05-31T10:45:33.047Z" level=info msg="finished streaming call with code NotFound" error="rpc error: code = NotFound desc = workflows.argoproj.io \"do-pipline-ztr2w\" not found" grpc.code=NotFound grpc.method=WatchWorkflows grpc.service=workflow.WorkflowService grpc.start_time="2021-05-31T10:45:33Z" grpc.time_ms=4.47 span.kind=server system=grpc
time="2021-05-31T10:45:36.187Z" level=info msg="finished streaming call with code Canceled" error="rpc error: code = Canceled desc = context canceled" grpc.code=Canceled grpc.method=WatchWorkflows grpc.service=workflow.WorkflowService grpc.start_time="2021-05-31T10:30:11Z" grpc.time_ms=925019 span.kind=server system=grpc
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x5a4482]
time="2021-05-31T10:45:33.047Z" level=info msg="finished streaming call with code NotFound" error="rpc error: code = NotFound desc = workflows.argoproj.io \"do-pipline-ztr2w\" not found" grpc.code=NotFound grpc.method=WatchWorkflows grpc.service=workflow.WorkflowService grpc.start_time="2021-05-31T10:45:33Z" grpc.time_ms=4.47 span.kind=server system=grpc
time="2021-05-31T10:45:36.187Z" level=info msg="finished streaming call with code Canceled" error="rpc error: code = Canceled desc = context canceled" grpc.code=Canceled grpc.method=WatchWorkflows grpc.service=workflow.WorkflowService grpc.start_time="2021-05-31T10:30:11Z" grpc.time_ms=925019 span.kind=server system=grpc
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x5a4482]

goroutine 29074 [running]:
bufio.(*Writer).Flush(0xc00115fb00, 0x1, 0x7f4e459ec698)
	/usr/local/go/src/bufio/bufio.go:607 +0x62
net/http.(*response).Flush(0xc0005089a0)
	/usr/local/go/src/net/http/server.go:1673 +0x35
github.com/argoproj/pkg/grpc/http.writeKeepalive(0x3442c00, 0xc0005089a0, 0xc002198088)
	/go/pkg/mod/github.com/argoproj/pkg@v0.8.1/grpc/http/forwarders.go:151 +0x173
github.com/argoproj/pkg/grpc/http.keepalive(0x344b840, 0xc000658a50, 0x3442c00, 0xc0005089a0, 0xc002198088)
	/go/pkg/mod/github.com/argoproj/pkg@v0.8.1/grpc/http/forwarders.go:166 +0xa5
created by github.com/argoproj/pkg/grpc/http.withKeepalive
	/go/pkg/mod/github.com/argoproj/pkg@v0.8.1/grpc/http/forwarders.go:174 +0x9a
	
```
---
<!-- Issue Author: Don't delete this message to encourage other users to support your issue! -->
**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

--- a/go.mod
+++ b/go.mod
@@ -13,7 +13,7 @@ require (
 	github.com/aliyun/aliyun-oss-go-sdk v2.1.8+incompatible
 	github.com/antonmedv/expr v1.8.8
 	github.com/argoproj/argo-events v1.2.0
-	github.com/argoproj/pkg v0.8.1
+	github.com/argoproj/pkg v0.9.1
 	github.com/baiyubin/aliyun-sts-go-sdk v0.0.0-20180326062324-cfa1a18b161f // indirect
 	github.com/blushft/go-diagrams v0.0.0-20201006005127-c78c821223d9
 	github.com/colinmarc/hdfs v1.1.4-0.20180805212432-9746310a4d31

